haystack/pipelines/base.py

```diff
@@ -2313,6 +2313,15 @@ def send_pipeline_event_if_needed(self, is_indexing: bool = False):
             should_send_event = (
                 self._has_event_time_interval_exceeded() or self._has_event_run_total_threshold_exceeded()
             )
+            
+            # RayPipeline's components are not compatible with the serialization in `get_config()` called by
+            # `send_pipeline_event()`, so the event sending is suspended for RayPipelines (it would crash anyhow)
+            from haystack.pipelines.ray import RayPipeline
+
+            if isinstance(self, RayPipeline):
```

Comment:
Instead of checking the type here, I would add this function to the `RayPipeline` class:
```python
    def send_pipeline_event(self, is_indexing: bool = False):
        pass
```

Reply:
Thanks @masci, good idea, I have included this now.